### DISCUSSION TITLE
Blend2D support for non-modern compillers

### DISCUSSION THREAD

**Discussion by JouriM66** [blend2d/blend2d#129] [Ideas]
Library interface itself not needed any C++11 and more features from compiller and possible can be used with any "old" compiller.
I made some changes in interface headers to make it possible.
Now they can be used with Borland compiller which is C++98 std or even lower.
Here is fork of this repository [at my place](https://github.com/JouriM66/blend2d) (https://github.com/JouriM66/blend2d).
May be somebody found them usefull.

PS: Its my first repo and I have no idea how to publish these changes in other way :)

**Comment by kobalicek** [comment]
Hello, that's some effort!

However, I'm not sure I would want to maintain the support for C++ compilers that are more than 10 years old. I understand the reasons, but I know that I would not be able to maintain it and it's a bit limiting in further development of the C++ API. A compile-time macro that would disable C++ API would be most likely still okay to maintain, but support for such old compilers that don't even have `<stdint.h>`, for example, that's something that I would like to avoid (this can hopefully be fixed by users, to add this header in the path so the compiler can find it).

**Comment by JouriM66** [comment]
I never mean to add another task to you.
You already have lot of work which nobody can do and maintaining compillers usefull for very small amount of people its waste of your time and power.
I mean little different.
I will change Blend2D interface to allow old compillers (at least the one family I have) to use public library interface. I will try to make it without affecting how library sources compile or work now.
I ask you for only one thing: after I finish this little task, may be you take a look at my work and, if you cant find any magor problems with it, apply it to main branch.
This will allow users of "old crap" like me to use Blend2D interface in feature without spending several days to adapt it every time new version comes.
Whats all.

**Comment by JouriM66** [comment]
It seems I finished all basic changes.
Now library and all tests compiled and run successfully.
Borland can use interface headers.
Can you please take a look?
Rep:
https://github.com/JouriM66/blend2d

**Comment by kobalicek** [comment]
Hi, in my opinion this has no chance of being merged. It just changes too much in a direction that I would like not to go. It's really too much.

**Comment by JouriM66** [comment]
Ok.
It means I stay at my branch and forget about main one.
Let it be.
